src/aws/osml/image_processing/gdal_tile_factory.py

```diff
@@ -245,21 +247,45 @@ def find_appropriate_r_level(src_bbox, tile_width) -> int:
         map1 = src_x_interpolator(dst_x, dst_y).astype(np.float32)
         map2 = src_y_interpolator(dst_x, dst_y).astype(np.float32)
 
-        logger.debug(
-            f"Sanity check remap array sizes. " f"They should match the desired map tile size {tile_size[0]}x{tile_size[1]}"
+        self.logger.debug(
+            f"Sanity check remap array sizes. They should match the desired map tile size {tile_size[0]}x{tile_size[1]}"
         )
-        logger.debug(f"map1.shape = {map1.shape}")
-        logger.debug(f"map2.shape = {map2.shape}")
+        self.logger.debug(f"map1.shape = {map1.shape}")
+        self.logger.debug(f"map2.shape = {map2.shape}")
+
+        # Set the scalar for out of bounds pixels to 0 and bias the image just slightly such that all values
+        # of 0 in the output are only from out of bounds pixels.  Valid range for the scalar is 0-255 [int|tuple].
+        if src.ndim == 2:  # 1-band grayscale
+            scalar = 0
+            src[src == 0] = 1
+        else:  # 3-band color
```

Comment:
This is likely a bad assumption and some of the images with multiple bands may be complex SAR (2-bands) or multispectral imagery (4+ bands). We likely need to add band select and other preprocessing logic as part of the map-tile creation process. For now suggest tightening these checks and error gracefully if the assumptions are not met.